bciers/apps/registration/tests/components/form/widgets/URLWidget.test.tsx

```diff
@@ -0,0 +1,104 @@
+import { userEvent } from "@testing-library/user-event";
+import { render, screen } from "@testing-library/react";
+import { RJSFSchema } from "@rjsf/utils";
+import FormBase from "@/app/components/form/FormBase";
+// import { checkTextWidgetValidationStyles } from "@/tests/helpers/form";
+
+const url = "https://example.com";
+const urlFieldLabel = "URL test field";
+const urlLabelRequired = `${urlFieldLabel}*`;
+const urlErrorMessage =
+  "Please enter a valid website link, e.g. http://www.website.com, https://www.website.com";
+
+const urlFieldSchema = {
+  type: "object",
+  required: ["urlTestField"],
+  properties: {
+    urlTestField: { type: "string", format: "uri", title: urlFieldLabel },
+  },
+} as RJSFSchema;
+
+const urlFieldUiSchema = {
+  urlTestField: {
+    "ui:widget": "URLWidget",
+  },
+};
+
+describe("RJSF URLWidget", () => {
+  it("should render a URL field", async () => {
+    render(<FormBase schema={urlFieldSchema} uiSchema={urlFieldUiSchema} />);
+    expect(screen.getByLabelText(urlLabelRequired)).toBeVisible();
+  });
+
+  it("should be empty by default", async () => {
+    render(<FormBase schema={urlFieldSchema} uiSchema={urlFieldUiSchema} />);
+
+    const input = screen.getByLabelText(urlLabelRequired);
+
+    expect(input).toHaveValue("");
+  });
+
+  it("should render the URL value when formData is provided", async () => {
+    render(
+      <FormBase
+        schema={urlFieldSchema}
+        formData={{ urlTestField: url }}
+        uiSchema={urlFieldUiSchema}
+      />,
+    );
+
+    const input = screen.getByLabelText(urlLabelRequired);
+    expect(input).toHaveValue(url);
+  });
+
+  it("should allow entering a URL", async () => {
+    render(<FormBase schema={urlFieldSchema} uiSchema={urlFieldUiSchema} />);
+    const input = screen.getByLabelText(urlLabelRequired);
+    await userEvent.type(input, url);
+    expect(input).toHaveValue(url);
+  });
+
+  it("should trigger validation error for invalid URL", async () => {
+    render(<FormBase schema={urlFieldSchema} uiSchema={urlFieldUiSchema} />);
+    const input = screen.getByLabelText(urlLabelRequired);
+    await userEvent.type(input, "example.com");
```

Comment:
how hard/easy is it to change this behaviour? IMO, "example.com" should be considered a valid URL. As a naive user, I wouldn't expect to have to specify "https://www.example.com" - I don't consider that to be the norm anymore. 
I would call "example" or ".com" invalid URLs.

{end rant}

Reply:
Haha I pushed to create our own custom validation for this because I thought forcing users to add `http://` or `https://` before a url was silly. Though I got overridden pre-mvp. There is a [ticket](https://github.com/bcgov/cas-registration/issues/509) for it, it's out of scope for this PR.